kythe/rust/indexer/src/indexer/analyzers.rs

```diff
@@ -94,14 +97,19 @@ impl<'a> UnitAnalyzer<'a> {
         // later to emit nodes and create a HashMap between a file path and its digest
         let mut file_vnames = HashMap::new();
         let mut file_digests = HashMap::new();
+        let mut generated_file_names = HashMap::new();
         for required_input in unit.get_required_input() {
             let analysis_vname = required_input.get_v_name();
             let storage_vname: VName = analysis_to_storage_vname(analysis_vname);
             let info = required_input.get_info();
-            let path = info.get_path().to_owned();
+            let source_path = info.get_path().to_owned();
+            let processed_path = storage_vname.get_path().to_owned();
```

Comment:
It's possible for more than one file to have the same "processed_path" even without generated files.  The "source_path" should be unique, however.

Reply:
The processed path is based on the source path, and the processed path is equal to the path inside the save_analysis file. The main reason for this PR overall is that the paths in the save_analysis file for generated files does not match up with the source_file path.